File: java/two-fer/src/test/java/TwoferTest.java
import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

public class TwoferTest {

    private Twofer twofer;

    @BeforeEach
    public void setup() {
        twofer = new Twofer();
    }

    @Test
    public void noNameGiven() {
        String input = null;
        String expected = "One for you, one for me.";

        assertEquals(expected, twofer.twofer(input));
    }

    @Test
    public void aNameGiven() {
        String input = "Alice";
        String expected = "One for Alice, one for me.";

        assertEquals(expected, twofer.twofer(input));
    }

    @Test
    public void anotherNameGiven() {
        String input = "Bob";
        String expected = "One for Bob, one for me.";

        assertEquals(expected, twofer.twofer(input));
    }
}

File: kotlin/two-fer/src/test/kotlin/TwoferTest.kt
import org.junit.jupiter.api.Test
import org.amshove.kluent.`should be equal to`

class TwoferTest {

    @Test
    fun noNameGiven() {
        twofer() `should be equal to` "One for you, one for me."
    }

    @Test
    fun aNameGiven() {
        twofer("Alice") `should be equal to` "One for Alice, one for me."
    }

    @Test
    fun anotherNameGiven() {
        "One for Bob, one for me." `should be equal to`  twofer("Bob")
    }

    @Test
    fun emptyStringGiven() {
        "One for , one for me." `should be equal to`  twofer("")
    }

}
